qa/tasks/util/__init__.py

```diff
@@ -165,12 +169,21 @@ def introspect_roles(ctx, logger, quiet=True):
                 non_storage_cluster_nodes += [remote.hostname]
             if role_type == 'osd':
                 nodes_storage += [remote.hostname]
+            if role_type == 'mon':
+                nodes_monitor += [remote.hostname]
             if role_type not in role_types[idx]:
                 role_types[idx] += [role_type]
         idx += 1
     nodes_cluster = list(set(nodes_cluster))
     nodes_gateway = list(set(nodes_gateway))
     nodes_storage = list(set(nodes_storage))
+    nodes_monitor = list(set(nodes_monitor))
+
+    nodes_random_storage = random.sample(nodes_storage, 2 if len(nodes_storage) > 1 else 1)
+    nodes_random_monitor = random.sample(nodes_monitor, 2 if len(nodes_monitor) > 1 else 1)
+    nodes_random_storage = list(set(nodes_random_storage))
+    nodes_random_monitor = list(set(nodes_random_monitor))
```

Comment:
Why do you need these two lines?

Reply:
Suppose because they are used for all other also?

Reply:
> Suppose because they are used for all other also?

could you please elaborate, I don't get

Reply:
To make it clear what I did not understand why do you do casting a list to set and then back to list?

Reply:
Because code was written like this before I started editing it.

Reply:
ah, I see, the similar code lines are created initially by Nathan above, but he left it by some reason alone... so you do not need copy it, because there is no sense in this kind double conversion here. Originally this 'list -> set -> list' conversion can be used to remove duplicates, for example:
```
>>> a=[3, 1, 1, 2]
>>> a
[3, 1, 1, 2]
>>> set(a)
set([1, 2, 3])
>>> list(set(a))
[1, 2, 3]
```
But here we already got rid of multiset earlier (multiset is a set with duplicate elements), so you do not need use it anymore.